### PR TITLE
Support for Ruby 3.0 (Fixes #1257)

### LINKED ISSUE #1257: Ruby 3.0 build failure
```
In file included from src/IceRuby/Connection.cpp:5:
In file included from src/IceRuby/Connection.h:8:
In file included from src/IceRuby/Config.h:68:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby.h:38:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/ruby.h:23:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/defines.h:73:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/backward/2/attributes.h:43:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/internal/attr/pure.h:25:
/usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/assert.h:132:1: error: '__declspec' attributes are not enabled; use '-fdeclspec' or '-fms-extensions' to enable support for __declspec attributes
RBIMPL_ATTR_NORETURN()
^
/usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/internal/attr/noreturn.h:29:33: note: expanded from macro 'RBIMPL_ATTR_NORETURN'
# define RBIMPL_ATTR_NORETURN() __declspec(noreturn)
                                ^
In file included from src/IceRuby/Connection.cpp:5:
In file included from src/IceRuby/Connection.h:8:
In file included from src/IceRuby/Config.h:68:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby.h:38:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/ruby.h:24:
In file included from /usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/internal/anyargs.h:77:
/usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/internal/intern/vm.h:64:1: error: '__declspec' attributes are not enabled; use '-fdeclspec' or '-fms-extensions' to enable support for __declspec attributes
RBIMPL_ATTR_NORETURN()
^
/usr/local/Cellar/ruby/3.0.0_1/include/ruby-3.0.0/ruby/internal/attr/noreturn.h:29:33: note: expanded from macro 'RBIMPL_ATTR_NORETURN'
# define RBIMPL_ATTR_NORETURN() __declspec(noreturn)
```

### PR DESCRIPTION
**Pull request by externl**

### CODE REVIEW

File: ruby/config/Make.rules [hunk 1]
```diff
@@ -14,6 +14,7 @@ RUBY ?= ruby
 ruby-call = $(shell $(RUBY) -e 'require "rbconfig"; puts RbConfig::expand("$1")')
 
 # Ruby compiler flags
+platform_cxx    := $(call ruby-call, $$(CXX))
```
Comment (pepone) [inline ruby/config/Make.rules:17]: Where is `platform_cxx` used?

Comment (pepone) [reply ruby/config/Make.rules:17]: I guess that is to use the same compiler as the ruby interpreter?

Comment (externl) [reply ruby/config/Make.rules:17]: It's used in this file. This is to override the compiler as reconfigure provides the compiler and some flags. 

https://github.com/zeroc-ice/ice/blob/3.7/config/Make.project.rules/#L1146

```
# Ruby 2.6
❯ /usr/bin/ruby -e 'require "rbconfig"; puts RbConfig::expand("$(CXX)")'
xcrun clang++

# Ruby 3.0
~/src/github.com/zeroc-ice/ice/3.7 ruby-3.0
❯ /usr/local/opt/ruby/bin/ruby -e 'require "rbconfig"; puts RbConfig::expand("$(CXX)")'
clang++ -fdeclspec

**Review by pepone — APPROVED** [review]
Looks good

**Pull request merged**